doc: update BM doc for 0.9.0 (#389)
nrfconnect/sdk-nrf-bm: doc/nrf-bm/release_notes/release_notes_0.9.0.rst

.. _nrf_bm_release_notes_090:

Release Notes for |BMlong| v0.9.0
#################################

.. contents::
   :local:
   :depth: 2

|BMshort| v0.9.0 is an early release of this product, demonstrating a set of samples for the `nRF54L15 DK`_ with build targets supporting the following SoCs:

* `nRF54L15`_
* `nRF54L10`_ (emulated on the nRF54L15 DK)
* `nRF54L05`_ (emulated on the nRF54L15 DK)

The supported DK version is `nRF54L15 DK v0.9.3`_.

> The release includes a prototype version of the S115 SoftDevice - a proprietary Bluetooth® LE stack to be used on the nRF54L15 DK.

Reviewer: ```suggestion
The release includes a prototype version of the S115 SoftDevice - a proprietary Bluetooth® LE stack and it comes in three variants targeting nRF54L15, L10, and L05.
```

S115 SoftDevice
***************

The S115 SoftDevice has not been updated since last release and its version is still ``s115_9.0.0-3.prototype``.
The SoftDevice comes in three variants to support different SoCs of the nRF54L Series: nRF54L15, nRF54L10, and nRF54L05.

* :file:`s115_9.0.0-3.prototype_nrf54l15_softdevice.hex`
* :file:`s115_9.0.0-3.prototype_nrf54l10_softdevice.hex`
* :file:`s115_9.0.0-3.prototype_nrf54l05_softdevice.hex`

For more details about the SoftDevice, see :ref:`s115_docs`.

For API documentation, see `S115 9.0.0-3.prototype API reference`_.

Changelog
*********

The following sections provide detailed lists of changes by component.

SoftDevice Handler
==================

* Added the :c:func:`nrf_sdh_ble_conn_handle_get` function.

* Updated:

  * The system initialization to initialize on application level.
  * The entropy source to use the CRACEN true random number generator instead of the CRACEN pseudorandom number generator.

Boards
======

* Boards must now select a SoftDevice sysbuild Kconfig in the `Kconfig.sysbuild` file e.g. :kconfig:option:`SB_CONFIG_SOFTDEVICE_S115`.

> 

Reviewer 2: Please add an entry for changes in https://github.com/nrfconnect/sdk-nrf-bm/pull/386.

Here is what I have. Feel free to modify if you see the need.
```suggestion
Logging
=======

* Fixed an issue where builds would fail if the :kconfig:option:`CONFIG_LOG` Kconfig option was disabled and the :kconfig:option:`CONFIG_LOG_BACKEND_BM_UARTE` Kconfig option was enabled.
```

     * The name of the :c:struct:`bm_zms_evt_t` ``ate_id`` field to :c:member:`bm_zms_evt_t.id`.
     * The type of :c:member:`bm_zms_evt_t.result` from ``uint32_t`` to ``int``.

   * Fixed an issue where some data was written incorrectly to storage if the data size was not a multiple of the program unit of 16 bytes.

* :ref:`lib_ble_conn_params` library:

   * Fixed an issue that caused the :kconfig:option:`CONFIG_BLE_CONN_PARAMS_INITIATE_DATA_LENGTH_UPDATE` Kconfig option to be always hidden.

Samples
=======

Bluetooth samples
-----------------

* Added the :ref:`ble_hids_keyboard_sample` and :ref:`ble_hids_mouse_sample` samples.

* :ref:`ble_hrs_sample` sample:

  * Added support for bonding and pairing.

* :ref:`ble_cgms_sample` sample:

  * Corrected the return type for the :c:func:`ble_bas_battery_level_update` function.

Known issues and limitations
============================

* Some issues are observed when using iPhone as the peer during testing of the Bluetooth samples.
* Some issues are observed when using Linux with the Bluetooth Low Energy app in nRF Connect for Desktop as the peer during testing of the Bluetooth samples.

> * There are no samples or libraries available to demonstrate or support the Bluetooth LE Pairing and bonding functionality in the SoftDevice.

Author: deleted

.. _nrf_bm_release_notes_090:

Release Notes for |BMlong| v0.9.0
#################################

.. contents::
   :local:
   :depth: 2

|BMshort| v0.9.0 is an early release of this product, demonstrating a set of samples for the `nRF54L15 DK`_ with build targets supporting the following SoCs:

* `nRF54L15`_

> * `nRF54L10`_ (emulated on the nRF54L15 DK

Reviewer 3: ```suggestion
* `nRF54L10`_ (emulated on the nRF54L15 DK)
```
